### PR TITLE
[NEW] Create I2C temperature sensor workshop

### PR DESCRIPTION
**Pull request by davidbeechey**
Sensors workshop task to fill in the gaps of the I2C temperature sensor implementation. (This one chosen in particular because we have a sensor and board to test on which can be used in the workshop.)

To Do:

- [x] Write up README with instructions

### CODE REVIEW

File: 04-i2c_temp_sensor/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+## Software Prerequisites
+
+Make sure that you have Rust and `probe-rs` installed on your computer. You can find the instructions on our [Getting Started with Rust Wiki page](https://github.com/Hyp-ed/hyped-2025/wiki/Getting-Started-with-Rust).
+
+## Fill in the Blanks!
+
+In this section, you will complete the code provided so that the STM32 microcontroller reads temperature data from the I2C temperature sensor. The code is divided into three parts:
+
+1. **Temperature reading logic**: In the `sensors/src/temperature/rs` file, you need to implement the `read` function that reads temperature data from the I2C temperature sensor. You can use the `HypedI2C` struct provided to communicate with the sensor, you can see the methods available in the `io/src/i2c.rs` file. (You will need to read the datasheet for the sensor in order to work out how to read the temperature data.) All of the required register addresses are provided in the file, and may give you a hint as to how to read the temperature data. Make sure that you return the temperature data in degrees Celsius!
```
Comment (TomLonergan03) [inline 04-i2c_temp_sensor/README.md:42]: `sensors/src/temperature.rs`

File: 04-i2c_temp_sensor/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+2. **Initialise the temperature sensor**: In the `board/src/mainr.rs` file, you need to write the code to create a new `Temperature` struct. Have a look at the parameters of the `Temperature::new` function in the `sensors/src/temperature.rs` file to see what you need to pass in.
```
Comment (TomLonergan03) [inline 04-i2c_temp_sensor/README.md:44]: `board/src/main.rs`

**Review by TomLonergan03 — COMMENTED** [review]
looks pretty good, should be helpful for new people

**Review by licornes-fluos — APPROVED** [review]
Looks very interesting

**Pull request merged**